cpp/src/rolling/rolling.cu

```diff
@@ -367,5 +371,59 @@ std::unique_ptr<column> rolling_window(column_view const &input,
                                                     user_defined_aggregator, op, output_type, mr, 0);
 }
 
+std::unique_ptr<column> rolling_window(column_view const& input,
+                                       rmm::device_vector<cudf::size_type> const& group_offsets,
+                                       size_type preceding_window,
+                                       size_type following_window,
+                                       size_type min_periods,
+                                       rolling_operator op,
+                                       rmm::mr::device_memory_resource* mr)
+{
```

Comment:
The logic of this function should go in the new `groupby::windowed_aggregate`. There shouldn't be a need to add a new `rolling_window` public API.

Reply:
@jrhemstad, @devavret, I got to spend a little more time on refactoring this.

To move the logic in my grouped `rolling_window` overload into `windowed_aggregate()`, I would need access to `cudf::detail::rolling_window()` (and transitively, `dispatch_rolling()`, `aggregation_dispatcher`, and the rest). Should all of this now be moved to `rolling_detail.hpp`?

Reply:
That is correct. All detail APIs should've been declared in a corresponding header in `include/cudf/detail` for precisely this reason: to be used internally by other libcudf APIs. We might've missed the `detail/rolling.hpp` but it needs to exist. If you need help, I can make it for you.

Reply:
Thanks, mate. I think I have it going internally. I'll post here, as soon as is viable.

Reply:
Given how much has changed on `branch-0.12`, and that this PR was itself based on a WIP-branch, would it be alright if I could close this one, and raise an updated PR against `branch-0.13`? Would that be completely out of form?

Reply:
Not really, if you're discarding all the work done in this one. If you're just updating your code to work with branch-0.13, you should still use this PR.

Reply:
Thank you, @devavret. The changes are pretty significant, given that:
1. The old `rolling_window()` commits are being undone, in order to move it all to `groupby`. Those commits are hard to reconcile.
2. Also, I'd be refactoring the `detail` `rolling_window` code into a separate header, for reuse.

Thanks, mate. This should save a bit of time.